eng/native/gen-buildsys.sh

```diff
@@ -92,8 +92,10 @@ if [[ "$scan_build" == "ON" && -n "$SCAN_BUILD_COMMAND" ]]; then
     cmake_command="$SCAN_BUILD_COMMAND $cmake_command"
 fi
 
-if [[ "$host_arch" == "wasm" ]]; then
+if [[ "$host_arch" == "wasm" && "$2" =~ "-Browser-" ]]; then
     cmake_command="emcmake $cmake_command"
+elif [[ "$host_arch" == "wasm" && "$2" =~ "-WASI-" ]]; then
```

Comment:
instead of looking at a substring of the directory I'd prefer if we add another argument to this script (similar to `$scan_build`)

Reply:
I added new mandatory parameter for targetOS for the scripts.